parse_test.go

```diff
@@ -327,7 +328,22 @@ func TestParseSimpleMessage(t *testing.T) {
 			actual, err := Parse(tt.input)
 			require.NoError(t, err)
 
-			require.Equal(t, tt.expected, actual)
+			// Check that AST message is equal to expected one.
+			require.Equal(t, tt.expected, actual.Message)
+
+			// Check that AST message converted back to string is equal to input.
+
+			// Edge case: scientific notation number is converted to normal notation, hence comparison is bound to fail.
```

Comment:
I wonder if there is any way to preserve the notation

Reply:
in token we store string representation of numberLIteral
in node we store float representation of numberLiteral, using json.Unmarshal

so json.Unmarshal converts "1e3" to 1000.0

going back to string we don't have any information about the original notation.

So simple answer: it is not possible

Reply:
Maybe we could store it as a string in node?